cpp/DataStorm/sampleFilter/Reader.cpp

```diff
@@ -4,40 +4,56 @@
 #include <Ice/Ice.h>
 
 #include <iostream>
-#include <optional>
 #include <string>
 
+#include "TemperatureRange.h"
+
 using namespace std;
 
+void
+printSample(const DataStorm::Sample<string, float>& sample)
+{
+    cout << "sample: key=" << sample.getKey() << ", value=" << sample.getValue() << std::endl;
+}
+
+void
+printSamples(const vector<DataStorm::Sample<string, float>>& samples)
+{
+    for (const auto& sample : samples)
+    {
+        printSample(sample);
+    }
+}
+
 int
 main(int argc, char* argv[])
 {
-    try
-    {
-        // Instantiates DataStorm node.
-        DataStorm::Node node{argc, argv};
+    // CtrlCHandler is a helper class that handles Ctrl+C and similar signals. It must be constructed at the beginning
+    // of the program, before creating a DataStorm node or starting any thread.
+    Ice::CtrlCHandler ctrlCHandler;
 
-        // Instantiates the "hello" topic. The topic uses strings for keys and values.
-        DataStorm::Topic<string, string> topic{node, "hello"};
+    // Instantiates DataStorm node.
+    DataStorm::Node node{argc, argv};
 
-        // Configure readers to never clear the history. We want to receive all the samples written by the writers.
-        topic.setReaderDefaultConfig({std::nullopt, std::nullopt, DataStorm::ClearHistoryPolicy::Never});
+    ctrlCHandler.setCallback(
+        [&ctrlCHandler, &node](int)
```

Comment:
Was reviewing your other demo and realized that my suggestion was incomplete.
You can also remove the capture of `ctrlCHandler` now, since it isn't being set to `nullptr` anymore:
```suggestion
        [&node](int)
```